force-app/main/default/classes/OpportunityTriggerHelper.cls

```diff
@@ -0,0 +1,217 @@
+/**
+* @description       : 
+* @author            : Nazar Voinarovych
+* @group             : 
+* @last modified on  : 10-07-2024
+* @last modified by  : Nazar Voinarovych
+**/
+public with sharing class OpportunityTriggerHelper {
+    
+    public static void updatePlannedSalesWhenOpportunityClosedWon(){
+        try{
+            
+            List<Opportunity> updatedOpportunities=[SELECT Id,CloseDate,StageName,Amount,(SELECT Id,Product2.Condition__c FROM OpportunityLineItems) FROM Opportunity 
+                                                    WHERE StageName='Closed Won' AND Id IN :Trigger.New];
+            
+            Map<Integer,Integer> monthAndYear=new Map<Integer,Integer>();
+            
+            Map<String,Planned_Sales__c> plannedSales=new Map<String,Planned_Sales__c>();
+            
+            Map<Id,Planned_Sales__c> plannedSalesToUpdate=new Map<Id,Planned_Sales__c>();
+            
+            if(!updatedOpportunities.isEmpty()){
+                for(Opportunity opportunity:updatedOpportunities){
+                    
+                    monthAndYear.put(opportunity.CloseDate.month(),opportunity.CloseDate.year());
+                    
+                }
+                
+                if(!monthAndYear.isEmpty()){
+                    for(Planned_Sales__c plannedSale:[SELECT Id,Year__c,Month__c,Car_Type__c,Total_Amount_Earned__c,Count_of_Cars_Sold__c 
+                                                      FROM Planned_Sales__c 
+                                                      WHERE Year__c IN:monthAndYear.values()
+                                                      AND Month__c IN:monthAndYear.keySet()]){
+                                                          
+                                                          String key=plannedSale.Year__c+'_'+plannedSale.Month__c+'_'+plannedSale.Car_Type__c;
+                                                          plannedSales.put(key,plannedSale); 
+                                                      }
+                    
+                    if(!plannedSales.isEmpty()){
+                        for(Opportunity opportunity:updatedOpportunities){
+                            String key1=opportunity.CloseDate.year()+'_'+opportunity.CloseDate.month();
+                            
+                            for(OpportunityLineItem opp:opportunity.OpportunityLineItems){
+                                String key2=key1+'_'+opp.Product2.Condition__c;
+                                
+                                if(plannedSales.containsKey(key2)){
+                                    Planned_Sales__c plannedSale=plannedSales.get(key2);
+                                    
+                                    if(plannedSale!=null){
+                                        plannedSale.Total_Amount_Earned__c+=opportunity.Amount;
```

Comment:
Adding Opportunity line item amount on each LineItem iteration will not calculate if correctly

If you have 10 cars each 100k -> Opp.Amount = 1 mil, 10 iterations will add 10 mil instead of 1 min

Another thing with Count_of_Cars_Sold__c. You have to only add cars with same Condition as Car Type on planned sales